Lib/test/test_coroutines.py

```diff
@@ -2342,12 +2342,24 @@ def test_unawaited_warning_when_module_broken(self):
         orig_wuc = warnings._warn_unawaited_coroutine
         try:
             warnings._warn_unawaited_coroutine = lambda coro: 1/0
-            with support.captured_stderr() as stream:
-                corofn()
-                support.gc_collect()
-            self.assertIn("Exception ignored in", stream.getvalue())
-            self.assertIn("ZeroDivisionError", stream.getvalue())
-            self.assertIn("was never awaited", stream.getvalue())
+            try:
+                # Note: catch_unraisable_exception() resurect the coroutine
+                with support.catch_unraisable_exception() as cm, \
+                     support.captured_stderr() as stream:
+                    # only store repr() to avoid keeping the coroutine alive
+                    coro = corofn()
+                    coro_repr = repr(coro)
+
+                    # clear reference to the coroutine without awaiting for it
+                    del coro
+                    support.gc_collect()
+
+                self.assertEqual(repr(cm.unraisable.object), coro_repr)
+                self.assertEqual(cm.unraisable.exc_type, ZeroDivisionError)
+                self.assertIn("was never awaited", stream.getvalue())
```

Comment:
It's a little bit strange to have to catch half of the error message from stderr, and the other half from sys.unraisablehook. My PR #13488 will allow to inject the error message in the unraisable exception to catch both at the same time.